neurovault/apps/statmaps/tasks.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import
+
+import os
+import numpy
+import pandas as pd
+from celery import shared_task
+import nibabel as nib
+from nilearn.plotting import plot_glass_brain
+from nilearn.image import resample_img
+from neurovault.apps.statmaps.models import Image
+
+@shared_task
+def generate_glassbrain_image(nifti_file,pk):
+  nifti_file = str(nifti_file)
+  png_img_name = "glass_brain_%s.png" % pk
+  png_img_path = os.path.join(os.path.split(nifti_file)[0],png_img_name)
+  glass_brain = plot_glass_brain(nifti_file)
+  glass_brain.savefig(png_img_path)
+
+
+@shared_task
+def make_correlation_df(resample_dim=[4,4,4],pkl_path="/opt/image_data/matrices/pearson_corr.pkl"):
+  # Get standard space brain
+  reference = os.path.join(os.environ['FREESURFER_HOME'],'subjects', 'fsaverage', 'mri', 'brainmask.nii.gz')
+  public_images = Image.objects.filter(collection__private=False)
+  # Get all image paths
+  image_paths = [image.file.path for image in public_images]
+  images_resamp, reference_resamp = resample_multi_images_ref(image_paths,reference,resample_dim)
+  # Read into pandas data frame
+  corr_df = pd.DataFrame()
+  row_count = 0
+  for image in images_resamp:
+    corr_df[row_count] = image.get_data().flatten()
+    row_count = row_count + 1
+  corr_df = corr_df.corr(method="pearson")
```

Comment:
Wouldn't that mean, given a vector of pearson correlations for image A against B, C, D.. Z - we wouldn't be comparing apples and oranges? For example, in an extreme case, image D has one non zero voxel. If we do an intersection, then we are comparing just that one voxel, and the similarity score could be ridiculously high. Our search would be biased to return more sparse maps, because we can't really control if users input thresholded maps.  

I think what you are hinting at (and the ideal) is to have a smarter comparison strategy that takes into account the intersection, size, etc, and I am in agreement, but I haven't really started to think about this yet (it goes well with aim 1!)  If the implicit mask is considered a standard and you say "Vanessa, just do it," then I definitely will go for it! But if people are sort of ambivalent about it, maybe we should do what is silly and straightforward... a correlation from all voxels within the same brain mask? Let me know your thoughts.